packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1863,6 +1923,20 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
       this.validateSamlAuthenticationOptions(props.fineGrainedAccessControl?.samlAuthenticationOptions);
     }
 
+    if (props.capacity?.nodeOptions) {
+      // Validate coordinator node configuration
+      const coordinatorConfig = props.capacity.nodeOptions.find(opt => opt.nodeType === NodeType.COORDINATOR)?.nodeConfig;
```

Comment:
I think `find` only returns the first matching item, what about the rest, are we going to do validation?

Reply:
We can't have more than one nodeConfig object for a single node type